### ISSUE TITLE
Refactoring to pass linting rules

### ISSUE THREAD

**Issue by s3alfisc** [s3alfisc/pyfixest#307]
@NKeleher has suggested the following linting rules: 

```
select = [
  "F", # Pyflakes
  "E", # pycodestyle errors
  "W", # pycodestyle warnings
  "I", # isort
  "D", # flake8-docstrings
  "UP", # pyupgrade
  "SIM", # flake8-simplify
  "TRY", # tryceratops
]
```

and suggests to break up the refactoring into module specific PRs which we merge into [fix-linting-errors](https://github.com/s3alfisc/pyfixest/tree/fix-linting-errors) step-by-step. 

- [ ]     utils
- [ ]     demean
- [ ]     model_matrix_fixest
- [ ]     feols
- [ ]     fepois
- [ ]     did/*
- [ ]     summarize
- [ ]     visualize
- [ ]     FixestMulti - mostly docstyle and upgrade
- [ ]     FormulaParser - mostly docstyle and upgrade
- [ ]     estimation - mostly docstyle and exception handling

Then when all of these are linted, we merge fix-linting-errors to the master branch.

**Comment by NKeleher** [comment]
Perfect! Thanks @s3alfisc. I've started working these items.

**Comment by s3alfisc** [comment]
As I have merged all changes to master with [5408f25](https://github.com/s3alfisc/pyfixest/commit/5408f256fdb2dfac93a0a6bde66acd45bf6502a5), this can be closed! Thanks for all your work on this @NKeleher 🎉
